cmd/artifact/info/info.go

```diff
@@ -79,12 +85,21 @@ func (o *artifactInfoOptions) RunArtifactInfo(ctx context.Context, args []string
 		if err != nil {
 			return err
 		}
+		resolvedRefs = append(resolvedRefs, ref)
+		resolvedRegs[reg] = true
+	}
 
-		client, err := utils.ClientForRegistry(ctx, reg, o.PlainHTTP, o.Printer)
-		if err != nil {
-			return err
-		}
+	// make sure we are logged into all relevant registries before we begin
+	if err := login.PerformAuthsFromConfigWithMap(ctx, authn.NewClient(), resolvedRegs); err != nil {
```

Comment:
Yes I added the autologin after doing this part of the code. We can move all the explicit login logic into the clients autologin function hand have it ON by default. 

i.e., i orginally added the autologin function because some of the commands required dependencies to be resolved so it was not possible to reliably get a full list of registries that are going to be accessed before starting.

Reply:
For now I updated the PR to use autologin everywhere but kept the boolean flag in the `Clietn(autologin bool)` function. If you want to make this the standard behavior we could remove that boolean flag and just have autologin be on always when creating a client via this utils function. If needed you can still create a client in a non standard config by using [NewClient](https://github.com/falcosecurity/falcoctl/pull/299/files#diff-d8403439d3fcd9c9295f349f664406118185d499b1d3c4d6f55ad90278b17d3bR40) directly instead.

Reply:
By removing the boolean flag we have two cases:
1. commands that exit after they complete such as `pull, push, install, etc...`, will execute the `autologin` logic for the registries that have credentials configured( config file or environment variables). If not, they will just use the credentials already configured using the `login` command. Anyway, the overhead of the `autologin` is minimal in my opinion;
2. commands such as `follow` will run only once the `autologin` logic, and then cache the `credentialLogic` for later use.

I would make the autologin the standard behavior as you suggested, what do you think?

Reply:
Yeah its a small overhead I agree it should be fine to make it standard behavior for the client.